### DISCUSSION TITLE
How I got AutoOrtho running on windows 11, without the read-only trick

### DISCUSSION THREAD

**Discussion by chezhj** [kubilus1/autoortho#395] [Show and tell]
After very long period of testing and trying I finally can say I've got AutoOrtho (latest) running smoothly

* I need both Dokan an Winfsp to be installed
* Remove all previous AutoOrtho directories from you "Custom Scenery" as there might be links that are not used and cause error
* Make sure "warnings on missing scenery files" is unchecked
* DISABLE any antivirus and realtime checking as it causes delays and makes it difficult for X-Plane to read scenery
* The junctions made by AO, are handled strange by X-Plane. They are seen as new scenery, an put on top of the scenery_pack file

**Install** 

- unzip autoortho_win_*.zip to a directory of your choice
- run the exe (I ran as admin, but don't think that is needed)
- set directories & cache
- install scenery and WAIT
- If finished, run and then exit (ctrl-c)

* run X-Plane(without running AO)
* from the main menu, quit
* modify scenery_packs.ini 

```
SCENERY_PACK Custom Scenery/yAutoOrtho_Overlays/
SCENERY_PACK Custom Scenery/z_ao_eur/
SCENERY_PACK Custom Scenery/z_ao_*/ <- any other scenery you have installed
SCENERY_PACK Custom Scenery/z_autoortho/
```
**Running**
* start X-Plane (without running AO)
* now start ao and run
* now choose your plane and airport 

**Stopping**
* Before you stop X-Plane, stop AO first, this will reset the junctions back to "normal" links, ensuring X-Plane won't mess with the scenery_packs.ini
![image](https://github.com/kubilus1/autoortho/assets/59411979/e65bb3ec-4563-4f50-afe0-6632d796baac)

**Update 21-09-2023**
Tested only once but it seems that SAM's automatic library adding to the scenery_pack is causing the z_ao* links to be inserted on top too. I've now configured SAM to insert libraries manually and the scenery_packs is stable, with AO running (or not)
This means that running X-plane sequence is
- start AO & run
- start X-Plane

**Comment by marloscaida** [comment]
Work fine! Thanks you.

**Comment by marloscaida** [comment]
I thought it had worked well for me, but after making several flights I realized that it doesn't charge my Orthophotos. This method definitely doesn't work for me. Thanks all the same.
Regards
carlos

**Reply by chezhj** [reply]
How do you know? You should see files being reference in de cmd window of AO.

**Comment by marloscaida** [comment]
Chezhj:
The AO window shows the same lines as before starting XP and remains that way throughout the flight. The landscape is the default of the simulator. Today I made two exactly the same flights to test the views. Short flight between Mendoza (SAME) and Santiago de Chile (SCEL).

With your method, AO doesn't work. It remains dormant, but does not detect the start and position of the aircraft.

However, when I start AO before the simulator, the AO window shows the activity of this software when I start XP.

I admit only one thing, which I did not do when following your instructions: I only have WinFSP and I do not have Dokan (which I uninstalled because I could never get it to work on my installation).

I'm not claiming that your instructions don't work, what I'm saying is that they don't work for me, with my particular installation. Carlos.

**Reply by chezhj** [reply]
That is a pity. Somehow your situation doesn't add up. Because when AO is installed (directories there and scenery_pack contains correct info) you should either see the AO-tiles or x-plane will say that it cant load.. If you want share xplane log and scenery_pack, I try to figure out what's going on

**Comment by marloscaida** [comment]
Later, I will send you the files for you to analyze. For me the problem is that with your AO method it does not detect the start of the simulator and therefore, it remains dormant waiting for it to start to take action.
Give me a few hours and I'll share my files.

**Comment by pntrafewl** [comment]
Your method is working great for me. Thanks for the write up.

**Comment by marloscaida** [comment]
[Metodo 3 INI.zip](https://github.com/kubilus1/autoortho/files/12701751/Metodo.3.INI.zip)
[Metodo 2 marlos.zip](https://github.com/kubilus1/autoortho/files/12701754/Metodo.2.marlos.zip)
[Metodo 1 chezhj.zip](https://github.com/kubilus1/autoortho/files/12701768/Metodo.1.chezhj.zip)

Three methods to use AO:

The first method is the one proposed by Chezhj:

1.- XP12 is started and the aircraft is positioned at the airport (scenery_pack.ini does not have the read-only attribute).
2.- AO is started once step 1 is completed.
3.- It is verified that AO does not detect the simulator in operation. Does nothing.
4.- AO is closed orderly with [CTRL] + C

The second method is the one that works for me:
1.- AO is started until it is waiting for the simulator.
2.- XP12 starts but with scenery_pack.ini with a read-only attribute.
3.- XP12 reports that it cannot open and edit the scenario configuration file.
4.- Now AO detects the start of the simulator and starts working on the Orthophotos.
5.- AO is closed orderly with [CTRL] + C

Third method: XP12 destroys my custom scenarios:
1.- AO is started until it is waiting for the simulator.
2.- XP12 starts but with scenery_pack.ini unprotected. XP12 edits scenery_pack.ini and removes the FUSE folders.
3.- You can see that all the buildings are blown up.
4.- AO is closed orderly with [CTRL] + C

For each of these methods I provide 5 files (each in .zip format):
- Two screenshots.
- LOG.TXT
- autoortho.log
- scenery_pack.ini

Regards

carlos

**Reply by chezhj** [reply]
Okay did some checking:
- in "my method" you need to start x-plane without positioning the aircraft, then start AO, then choose & position aircraft
- The reason scenery_packs is destroyed is because (see edit) SAM is messing it up. Copy the position of the libraries manually into scenery_packs and configure sam for manual library management. Now you can just start AO, then X-Plane without the read-only
Hope that helps

**Comment by marloscaida** [comment]
Thank you very much for the help. I'm going to try it and tell you later.

**Comment by marloscaida** [comment]
Applause. His method works perfectly.
I eliminated the lines that the SAM PLUGIN must enter.
I removed the read-only condition from scenery_pack.ini
I booted the sim to the options screen, without loading the scenario.
Start the AO until it passes the test and goes into Standby.
Perfect. Now AO works wonders.
Thank you very much for your valuable input.

regards
carlos

**Comment by marloscaida** [comment]
I have a problem with this.
SAM is a product that is not fully adapted to XP12.
When I close the simulator, the EXIT option does not delete the lines:
SCENERY_PACK C:\Program Files (x86)\Steam\steamapps\common\X-Plane 12\Resources\plugins\SAM\lib\SAM_GlobalTrees/
SCENERY_PACK C:\Program Files (x86)\Steam\steamapps\common\X-Plane 12\Resources\plugins\SAM\lib\SAM_Library/
SCENERY_PACK C:\Program Files (x86)\Steam\steamapps\common\X-Plane 12\Resources\plugins\SAM\lib\SAM_Seasons/
SCENERY_PACK Custom Scenery/SAM_DeveloperPack/

The next time I use your method it will cause problems with AO...
Regards

**Reply by chezhj** [reply]
I guess I didn't explain good enough what I meant with make sure SAM links are set on manual, see [this ](https://www.4xplane.nl/documentation/xo-and-sam-shortcuts-like-aa_asdfer-in-custom-scenery/) for a full explanation
In short
- configure SAM for manual libraries (NOT automatically)
- copy the library links into scenery_pack.ini

**Comment by marloscaida** [comment]
You are absolutely right. I have finally finished understanding the method and I am already applying it. AO works very well, with SAM included manually and with scenery_pack.ini checked out so XP can inspect it without problems. Again thank you very much.

Best Regards
carlos

**Comment by erwin4711** [comment]
so to be short : SAM settting:
![bandicam 2023-09-27 21-36-56-885](https://github.com/kubilus1/autoortho/assets/6015730/bbf4ca98-5521-4a3a-90d6-35b624d2be35)
